tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -203,14 +203,17 @@ Status StorageManager::array_open_for_reads(
     std::vector<FragmentMetadata*>* fragment_metadata) {
   STATS_START_TIMER(stats::Stats::TimerType::READ_ARRAY_OPEN)
 
-  // Open array without fragments
+  // Open array without fragments async. This loads the array schema which
+  // is not needed in this function, so it can safely be loaded in parallel
+  // to loading the fragment metadata
   auto open_array = (OpenArray*)nullptr;
-  RETURN_NOT_OK_ELSE(
-      array_open_without_fragments(array_uri, enc_key, &open_array),
-      *array_schema = nullptr);
-
-  // Retrieve array schema
-  *array_schema = open_array->array_schema();
+  std::vector<ThreadPool::Task> load_array_schema_task;
+  load_array_schema_task.emplace_back(io_tp_->execute([&, this]() {
+    RETURN_NOT_OK_ELSE(
+        array_open_without_fragments(array_uri, enc_key, &open_array),
```

Comment:
This must complete before L233 because `load_fragment_metadata` depends on `open_array` which is set here. The `load_fragment_metadata` implementation depends on `open_array->array_schema();` so we can not refactor to defer the `load_array_schema` call in `array_open_without_fragments`. 

However, waiting on this on L231 might provide some parallelism with `load_consolidated_fragment_meta`.

Reply:
I got too greedy in parallelizing last night :). You are right and I've swapped the waiting up to before we load the fragment metadata but after do we the listing and loading of consolidated metadata.